_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
+}
+```
+#### Scenario 7 - Test assert equal xml - element value mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
 
-*TODO: is this information current?*
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abcd</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): abc abcd  :  mismatched element text ($expected= /t:test/code , $actual= /t:test/code )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+### Scenario 8 - Test assert equal JSON - same json node and values as json object
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+
+return
+try{
+	test:assert-equal-json($actual, $expected)
+} catch($err) {
+ $err
+}
+```
+### Scenario 9 - Test assert equal JSON - same json node and values as json document
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ $err
+}
+```
+### Scenario 10 - Test assert equal JSON - Node mismatch as json document and validate expected error message
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+let $_ := map:put($actual,"c",333)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+let $errorMessageExpected := "Assert Equal Json failed (ASSERT-EQUAL-JSON-FAILED): " || $actualJson || " " || $expectedJson
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
```

Comment:
I would not try catch, just explain to the reader that this test will fail because the actual JSON object has an additional property that does not exist in the expected JSON object.